### PR TITLE
Upgrade pallets to FRAMEv2

### PR DESCRIPTION
**Pull request by KiChjang**
Also did `rustfmt` on the pallet crates.

### CODE REVIEW

File: pallets/parachain-system/src/lib.rs [hunk 1]
```diff
@@ -447,418 +290,130 @@ decl_module! {
...
-			Ok(idx) => &relevant_messaging_state.egress_channels[idx].1,
-			Err(_) => return Err(SendHorizontalErr::NoChannel),
-		};
-		if data.len() as u32 > channel_meta.max_message_size {
-			return Err(SendHorizontalErr::TooBig);
+		#[pallet::weight((1_000, DispatchClass::Operational))]
+		fn sudo_send_hrmp_message(
+			origin: OriginFor<T>,
+			message: OutboundHrmpMessage,
+		) -> DispatchResultWithPostInfo {
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:399]:
```suggestion
		) -> DispatchResult {
```

File: pallets/parachain-system/src/lib.rs [hunk 2]
```diff
@@ -447,418 +290,130 @@ decl_module! {
...
-			vfp.relay_parent_number.saturating_sub(LastUpgrade::get());
-
-		if relay_blocks_since_last_upgrade <= cfg.validation_upgrade_frequency {
-			// The cooldown after the last upgrade hasn't elapsed yet. Upgrade is not allowed.
-			return None;
+		#[pallet::weight((1_000, DispatchClass::Operational))]
+		fn sudo_send_upward_message(
+			origin: OriginFor<T>,
+			message: UpwardMessage,
+		) -> DispatchResultWithPostInfo {
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:389]:
```suggestion
		) -> DispatchResult {
```

File: pallets/parachain-system/src/lib.rs [hunk 3]
```diff
@@ -447,418 +290,130 @@ decl_module! {
...
+			ensure_root(origin)?;
+			let _ = Self::send_upward_message(message);
+			Ok(().into())
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:392]:
```suggestion
			Ok(())
```

File: pallets/parachain-system/src/lib.rs [hunk 4]
```diff
@@ -447,418 +290,130 @@ decl_module! {
...
-		if data.len() as u32 > channel_meta.max_message_size {
-			return Err(SendHorizontalErr::TooBig);
+		#[pallet::weight((1_000, DispatchClass::Operational))]
+		fn sudo_send_hrmp_message(
+			origin: OriginFor<T>,
+			message: OutboundHrmpMessage,
+		) -> DispatchResultWithPostInfo {
+			ensure_root(origin)?;
+			let _ = Self::send_hrmp_message(message);
+			Ok(().into())
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:402]:
```suggestion
			Ok(())
```

File: pallets/parachain-system/src/lib.rs [hunk 5]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+	/// value, then [`PendingValidationFunction`] must have a real value, and
+	/// together will coordinate the block number where the upgrade will happen.
+	pub(super) type PendingRelayChainBlockNumber<T: Config> =
+		StorageValue<_, RelayChainBlockNumber>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn new_validation_function)]
+	/// The new validation function we will upgrade to when the relay chain
+	/// reaches [`PendingRelayChainBlockNumber`]. A real validation function must
+	/// exist here as long as [`PendingRelayChainBlockNumber`] is set.
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:460]:
```suggestion
	/// The new validation function we will upgrade to when the relay chain
	/// reaches [`PendingRelayChainBlockNumber`]. A real validation function must
	/// exist here as long as [`PendingRelayChainBlockNumber`] is set.
	#[pallet::storage]
	#[pallet::getter(fn new_validation_function)]
```

File: pallets/parachain-system/src/lib.rs [hunk 6]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
 
-/// tests for this pallet
-#[cfg(test)]
-mod tests {
-	use super::*;
+	#[pallet::storage]
+	/// We need to store the new validation function for the span between
+	/// setting it and applying it. If it has a
+	/// value, then [`PendingValidationFunction`] must have a real value, and
+	/// together will coordinate the block number where the upgrade will happen.
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:452]:
```suggestion
	/// We need to store the new validation function for the span between
	/// setting it and applying it. If it has a
	/// value, then [`PendingValidationFunction`] must have a real value, and
	/// together will coordinate the block number where the upgrade will happen.
	#[pallet::storage]
```

File: pallets/parachain-system/src/lib.rs [hunk 7]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+	#[pallet::getter(fn new_validation_function)]
+	/// The new validation function we will upgrade to when the relay chain
+	/// reaches [`PendingRelayChainBlockNumber`]. A real validation function must
+	/// exist here as long as [`PendingRelayChainBlockNumber`] is set.
+	pub(super) type PendingValidationFunction<T: Config> = StorageValue<_, Vec<u8>, ValueQuery>;
+
+	#[pallet::storage]
+	#[pallet::getter(fn validation_data)]
+	/// The [`PersistedValidationData`] set for this block.
+	pub(super) type ValidationData<T: Config> = StorageValue<_, PersistedValidationData>;
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:466]:
```suggestion
	/// The [`PersistedValidationData`] set for this block.
	#[pallet::storage]
	#[pallet::getter(fn validation_data)]
	pub(super) type ValidationData<T: Config> = StorageValue<_, PersistedValidationData>;
```

File: pallets/parachain-system/src/lib.rs [hunk 8]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+
+	#[pallet::storage]
+	/// Were the validation data set to notify the relay chain?
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:469]:
```suggestion
	/// Were the validation data set to notify the relay chain?
	#[pallet::storage]
```

File: pallets/parachain-system/src/lib.rs [hunk 9]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+	pub(super) type DidSetValidationCode<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+	#[pallet::storage]
+	/// The last relay parent block number at which we signalled the code upgrade.
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:473]:
```suggestion
	/// The last relay parent block number at which we signalled the code upgrade.
	#[pallet::storage]
```

File: pallets/parachain-system/src/lib.rs [hunk 10]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+
+	#[pallet::storage]
+	#[pallet::getter(fn relevant_messaging_state)]
+	/// The snapshot of some state related to messaging relevant to the current parachain as per
+	/// the relay parent.
+	///
+	/// This field is meant to be updated each block with the validation data inherent. Therefore,
+	/// before processing of the inherent, e.g. in `on_initialize` this data may be stale.
+	///
+	/// This data is also absent from the genesis.
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:484]:
```suggestion
	/// The snapshot of some state related to messaging relevant to the current parachain as per
	/// the relay parent.
	///
	/// This field is meant to be updated each block with the validation data inherent. Therefore,
	/// before processing of the inherent, e.g. in `on_initialize` this data may be stale.
	///
	/// This data is also absent from the genesis.
	#[pallet::storage]
	#[pallet::getter(fn relevant_messaging_state)]
```

File: pallets/parachain-system/src/lib.rs [hunk 11]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
-		type HrmpMessageHandlers = SaveIntoThreadLocal;
-	}
+	#[pallet::storage]
+	#[pallet::getter(fn host_configuration)]
+	/// The parachain host configuration that was obtained from the relay parent.
+	///
+	/// This field is meant to be updated each block with the validation data inherent. Therefore,
+	/// before processing of the inherent, e.g. in `on_initialize` this data may be stale.
+	///
+	/// This data is also absent from the genesis.
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:494]:
```suggestion
	/// The parachain host configuration that was obtained from the relay parent.
	///
	/// This field is meant to be updated each block with the validation data inherent. Therefore,
	/// before processing of the inherent, e.g. in `on_initialize` this data may be stale.
	///
	/// This data is also absent from the genesis.
	#[pallet::storage]
	#[pallet::getter(fn host_configuration)]
```

File: pallets/parachain-system/src/lib.rs [hunk 12]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+	pub(super) type HostConfiguration<T: Config> = StorageValue<_, AbridgedHostConfiguration>;
 
-	pub struct SaveIntoThreadLocal;
+	#[pallet::storage]
+	/// The last downward message queue chain head we have observed.
+	///
+	/// This value is loaded before and saved after processing inbound downward messages carried
+	/// by the system inherent.
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:501]:
```suggestion
	/// The last downward message queue chain head we have observed.
	///
	/// This value is loaded before and saved after processing inbound downward messages carried
	/// by the system inherent.
	#[pallet::storage]
```

File: pallets/parachain-system/src/lib.rs [hunk 13]
```diff
@@ -889,952 +444,493 @@ decl_error! {
...
+
+	#[pallet::storage]
+	/// HRMP channels with the given recipients are awaiting to be processed. If a `ParaId` is
+	/// present in this vector then `OutboundHrmpMessages` for it should be not empty.
+	pub(super) type NonEmptyHrmpChannels<T: Config> = StorageValue<_, Vec<ParaId>, ValueQuery>;
+
+	#[pallet::storage]
+	/// The number of HRMP messages we observed in `on_initialize` and thus used that number for
+	/// announcing the weight of `on_initialize` and `on_finialize`.
+	pub(super) type AnnouncedHrmpMessagesPerCandidate<T: Config> = StorageValue<_, u32, ValueQuery>;
```
Comment (bkchr) [inline pallets/parachain-system/src/lib.rs:529]:
```suggestion
	/// The message queue chain heads we have observed per each channel incoming channel.
	///
	/// This value is loaded before and saved after processing inbound downward messages carried
	/// by the system inherent.
	#[pallet::storage]
	pub(super) type LastHrmpMqcHeads<T: Config> =
		StorageValue<_, BTreeMap<ParaId, MessageQueueChain>, ValueQuery>;

	#[pallet::storage]
	pub(super) type PendingUpwardMessages<T: Config> =
		StorageValue<_, Vec<UpwardMessage>, ValueQuery>;

	/// Essentially `OutboundHrmpMessage`s grouped by the recipients.
	#[pallet::storage]
	pub(super) type OutboundHrmpMessages<T: Config> =
		StorageMap<_, Twox64Concat, ParaId, Vec<Vec<u8>>, ValueQuery>;

	/// HRMP channels with the given recipients are awaiting to be processed. If a `ParaId` is
	/// present in this vector then `OutboundHrmpMessages` for it should be not empty.
	#[pallet::storage]
	pub(super) type NonEmptyHrmpChannels<T: Config> = StorageValue<_, Vec<ParaId>, ValueQuery>;

	/// The number of HRMP messages we observed in `on_initialize` and thus used that number for
	/// announcing the weight of `on_initialize` and `on_finialize`.
	#[pallet::storage]
	pub(super) type AnnouncedHrmpMessagesPerCandidate<T: Config> = StorageValue<_, u32, ValueQuery>;
```

**Review by bkchr — APPROVED** [review]
Ty :)

File: pallets/parachain-system/src/lib.rs [hunk 14]
```diff
@@ -35,214 +35,56 @@ use cumulus_primitives_core::{
...
-		/// exist here as long as [`PendingRelayChainBlockNumber`] is set.
-		PendingValidationFunction get(fn new_validation_function): Vec<u8>;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
 
-		/// The [`PersistedValidationData`] set for this block.
-		ValidationData get(fn validation_data): Option<PersistedValidationData>;
+	#[pallet::config]
+	#[pallet::disable_frame_system_supertrait_check]
```
[the remark this answers is not in the document]
Comment (bkchr) [reply pallets/parachain-system/src/lib.rs:62]: https://github.com/paritytech/substrate/pull/8606

**Comment by KiChjang** [comment]
@bkchr Comments addressed.

**Comment by KiChjang** [comment]
@bkchr Uh oh, I don't know how to fix the compiler errors from CI... it looks like the `XcmExecutor::execute_xcm` method now takes an extra argument for the weight limit of the call, and I've no idea how to calculate that! It seems the only way here is to add it as a parameter to whichever function that calls `execute_xcm`.

**Comment by bkchr** [comment]
You had run `cargo update`. Which you should do.

Now we have merged some more important pr that conflicts with your stuff. However, I think that we now only need to update `parachain-system`. If you are still interested, I think the easiest would be to restart from 0. If not, it is also fine.

**Comment by KiChjang** [comment]
Sounds good, I'll start from scratch again and resubmit a PR soon.

**Pull request closed without merging**